src/brain-map.ts

```diff
@@ -0,0 +1,30 @@
+import { GameState } from './game-state';
+
+export class BrainMap {
+  state: GameState[] = new Array(20000);
+
+  constructor(initState: GameState[] = []) {
+    initState.forEach((move, i) => {
+      this.state[i] = move;
+    });
+  }
+
+  set(key: string, state: GameState) {
+    this.state[this.createHash(key)] = state;
```

Comment:
What happens when two different keys result in the same hash value? The ole hash collision problem.

Reply:
I was thinking about it and with our limited dataset is a hash collision a real concern?

Reply:
It's not worth it to test the speed, but wouldn't trust it going forward. We had a collision hit between two short words in the Java string hash implementation at Groupon. So it does happen. But it wouldn't have an impact on testing the speed of the solution.

Reply:
gotcha. ill make sure it is safe